gradlew fails to build Docker image - gradle-wrapper.jar broken?

#### Software and hardware versions 
Aggregate master (git clone)
Ubuntu 16.04.5 LTS
```
java -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-0ubuntu0.16.04.1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```
#### Problem description
* gradle/wrapper/gradle-wrapper.jar seems to be broken.
* Gradle's handling of `JAVA_HOME` seems broken, at least on Ubuntu, however that's not an Aggregate issue.

#### Steps to reproduce the problem
Follow the [Docker build](https://github.com/opendatakit/aggregate/blob/master/docs/build-and-run-a-docker-image.md).
* Setup local postgres database
* [Fix gradle's broken `JAVA_HOME`](https://stackoverflow.com/a/44168551/2813717) by running:
```
which javac >/dev/null 2>&1 || die "ERROR: no 'javac' command could be found in your PATH"
export JAVA_HOME=$(dirname $(dirname $(readlink -f $(which javac)  )))
```

* Run the gradle build step:
```
./gradlew clean dockerBuild -xtest -PwarMode=complete                                
Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain
```
However, [gradle/wrapper](https://github.com/opendatakit/aggregate/blob/master/gradle/wrapper) contains a gradle JAR. Naively running that JAR indicates it might be broken:
```
java -jar gradle/wrapper/gradle-wrapper.jar 
Error: Invalid or corrupt jarfile gradle/wrapper/gradle-wrapper.jar
```

#### Expected behavior
* Gradle builds the Docker image.

### Other information
Trying to rebuild gradle-wrapper.jar by running `gradle wrapper --stacktrace` fails with gradle v 1.2.
File wrapper.gradle contains:
```
task wrapper(type: Wrapper) {
   gradleVersion = '2.10' // version required
}
```
Updating gradle to 2.10, running `gradle wrapper` fails with:
```
Caused by: org.gradle.api.internal.MissingMethodException: Could not find method compileOnly() for arguments [{group=javax.servlet, name=javax.servlet-api, version=3.1.0}] on root project 'aggregate'.
```

We're excited about the upcoming Docker images! Our infrastructure team is pushing towards Docker-based deployments, so we're happy to battle test anything Docker based (once I can get it to build that is).

Following instructions now and running IntelliJ IDE on my fork dbca-wa/aggregate:
- the instructions say `Gradle project: odk-aggregate` should be `Gradle project: aggregate`
- some jars seem to be missing still - running `appStartWar` doesn't start a war, but still bombs out:
```
error: error reading /home/florianm/projects/ODK/aggregate/libs/odk-tomcatutil-1.0.1.jar; error in opening zip file
error: error reading /home/florianm/projects/ODK/aggregate/libs/gwt-google-maps-v3-1.0.1.jar; error in opening zip file
error: error reading /home/florianm/projects/ODK/aggregate/libs/gwt-visualization-1.1.2.jar; error in opening zip file
error: error reading /home/florianm/projects/ODK/aggregate/libs/odk-httpclient-gae-4.5.2-1.jar; error in opening zip file
error: error reading /home/florianm/projects/ODK/aggregate/libs/odk-tomcatutil-1.0.1.jar; error in opening zip file
error: error reading /home/florianm/projects/ODK/aggregate/libs/gwt-google-maps-v3-1.0.1.jar; error in opening zip file
error: error reading /home/florianm/projects/ODK/aggregate/libs/gwt-visualization-1.1.2.jar; error in opening zip file
error: error reading /home/florianm/projects/ODK/aggregate/libs/odk-httpclient-gae-4.5.2-1.jar; error in opening zip file
/home/florianm/projects/ODK/aggregate/src/main/java/org/opendatakit/common/utils/gae/GaeHttpClientFactoryImpl.java:24: error: package org.opendatakit.http.conn does not exist
```

--- a/docs/build-and-run-a-docker-image.md
+++ b/docs/build-and-run-a-docker-image.md
@@ -9,7 +9,7 @@
   - Only then clone your fork of Aggregate - included .jar files only download correctly with Git LFS. See [Getting the code](https://github.com/opendatakit/aggregate#getting-the-code).
   - Install IntelliJ IDEA and import project as instructed to let IDEA download, build, and configure sources. See [Running the project - Import](https://github.com/opendatakit/aggregate#import)
 
-Note: this is a gentle reminder to follow the main [README](https://github.com/opendatakit/aggregate), not a comprehensive setup guide in itself.
+  *Note*: this is a gentle reminder to follow the main [README](https://github.com/opendatakit/aggregate), not a comprehensive setup guide in itself.
 
 ## Quick start - External PostgreSQL instance
 

--- a/docs/build-and-run-a-docker-image.md
+++ b/docs/build-and-run-a-docker-image.md
@@ -6,8 +6,10 @@
 - Already have a PostgreSQL 9.6 database running
 - Follow the instructions of the main [README](https://github.com/opendatakit/aggregate), including:
   - Install Git LFS, as well as its git hooks, and enable the git hooks for Git LFS.
-  - Only then clone your fork of Aggregate - included .jar files only download correctly with Git LFS.
-  - Install IntelliJ IDEA and import project as instructed to let IDEA download, build, and configure sources.
+  - Only then clone your fork of Aggregate - included .jar files only download correctly with Git LFS. See [Getting the code](https://github.com/opendatakit/aggregate#getting-the-code).
+  - Install IntelliJ IDEA and import project as instructed to let IDEA download, build, and configure sources. See [Running the project - Import](https://github.com/opendatakit/aggregate#import)
+
+Note: this is a gentle reminder to follow the main [README](https://github.com/opendatakit/aggregate), not a comprehensive setup guide in itself.
 
 ## Quick start - External PostgreSQL instance
 

